src/NativeShell/src/cordova/NativeShell.java

```diff
@@ -0,0 +1,188 @@
+                return disableFullscreen();
+            case "updateMediaSession":
+                return updateMediaSession();
+            case "hideMediaSession":
+                return hideMediaSession();
+            default:
+                callbackContext.error("unrecognized: " + action);
+                return false;
+        }
+    }
+
+    public boolean getDeviceInformation() {
+        // replace this later with a randomly generated persistent string stored in local settings
+        String deviceId =  Settings.Secure.getString(cordova.getActivity().getContentResolver(), Settings.Secure.ANDROID_ID);
+        String deviceName = Build.MODEL;
+        String appName = "Jellyfin Mobile";
+        String appVersion = Integer.toString(BuildConfig.VERSION_CODE);
+
+        JSONObject jsonObject = new JSONObject();
+        try {
+            jsonObject.put("deviceId", deviceId);
+            jsonObject.put("deviceName", deviceName);
+            jsonObject.put("appName", appName);
+            jsonObject.put("appVersion", appVersion);
+        } catch(Exception e) {
+            callbackContext.error("error: " + e.getMessage());
+            return false;
+        }
+
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, jsonObject);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean enableFullscreen() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                int visibility = SYSTEM_UI_FLAG_FULLSCREEN
+                        | SYSTEM_UI_FLAG_IMMERSIVE_STICKY
+                        | SYSTEM_UI_FLAG_HIDE_NAVIGATION;
+                cordova.getActivity().getWindow().getDecorView().setSystemUiVisibility(visibility);
+                cordova.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean disableFullscreen() {
+        cordova.getActivity().runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                cordova.getActivity().getWindow().getDecorView().setSystemUiVisibility(SYSTEM_UI_FLAG_VISIBLE);
+                cordova.getActivity().getWindow().addFlags(WindowManager.LayoutParams.FLAG_FORCE_NOT_FULLSCREEN);
+                cordova.getActivity().getWindow().clearFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN);
+            }
+        });
+        PluginResult pluginResult = new PluginResult(PluginResult.Status.OK);
+        callbackContext.sendPluginResult(pluginResult);
+        return true;
+    }
+
+    public boolean updateMediaSession() {
+        String action;
```

Comment:
cannot you declare all those on one line? looks too much waste of screen estate to me